crates/sui-move-build/tests/toolchain_version_tests.rs

```diff
@@ -0,0 +1,74 @@
+use std::path::PathBuf;
+
+use move_compiler::editions::{Edition, Flavor};
+use sui_move_build::{BuildConfig, CompiledPackage};
+
+#[test]
+fn no_toolchain_version() {
```

Comment:
These tests go via `build_config.build` so that they are end-to-end wrt the build process. Later they will grow to include writing toolchain version to `Move.lock`